### PR TITLE
Refactor auth functions that use localStorage logic into AuthAPIClient

### PR DESCRIPTION
**Pull request by emilyslouie**
## Notion ticket link

[Refactor token storage logic out of auth components in GraphQL frontend client](https://www.notion.so/uwblueprintexecs/Refactor-token-storage-logic-out-of-auth-components-in-GraphQL-frontend-client-7d8bed2ef34e415592fe24fb76673d39)

## Implementation description
* Removed `localStorage` functionality from UI components and moved it to `AuthAPIClient` by passing necessary parameters to these new helper functions
* Added GraphQL tags to the required functions in `AuthAPIClient` to replace functions if REST is chosen instead

## Steps to test
1. Run the app
2. Log in successfully
3. Refresh credentials successfully
4. Logout successfully

## What should reviewers focus on?
* the added functions in `AuthAPIClient`
* the scrubber tags to ensure the correct areas are tagged

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/APIClients/AuthAPIClient.ts
Comment (sherryhli) [inline frontend/src/APIClients/AuthAPIClient.ts:18]: Minor thing but let's type the function more specifically, you can import the missing types (e.g. `MutationFunctionOptions` from `@apollo/client`.

```ts
loginFunction: (
    options?:
      | MutationFunctionOptions<
          { login: AuthenticatedUser },
          OperationVariables
        >
      | undefined,
  ) => Promise<
    FetchResult<
      { login: AuthenticatedUser },
      Record<string, any>,
      Record<string, any>
    >
  >
```

File: frontend/src/APIClients/AuthAPIClient.ts
Comment (sherryhli) [inline frontend/src/APIClients/AuthAPIClient.ts:33]:
```ts
logoutFunction: (
    options?:
      | MutationFunctionOptions<
          {
            logout: null;
          },
          OperationVariables
        >
      | undefined,
  ) => Promise<
    FetchResult<
      {
        logout: null;
      },
      Record<string, any>,
      Record<string, any>
    >
  >
```

File: frontend/src/APIClients/AuthAPIClient.ts
Comment (sherryhli) [inline frontend/src/APIClients/AuthAPIClient.ts:46]:
```ts
refreshFunction: (
    options?:
      | MutationFunctionOptions<
          {
            refresh: string;
          },
          OperationVariables
        >
      | undefined,
  ) => Promise<
    FetchResult<
      {
        refresh: string;
      },
      Record<string, any>,
      Record<string, any>
    >
  >
```

**Review by sherryhli — CHANGES_REQUESTED** [review]
Great work! A couple of small changes and then you're good to go. I think there'll be some merge conflicts with #42, probably the easiest to wait until #42 is merged and then resolve them here

File: frontend/src/APIClients/AuthAPIClient.ts
Comment (emilyslouie) [reply frontend/src/APIClients/AuthAPIClient.ts]: thanks, made the changes 👍 and will wait till 42 is merged first

**Review by sherryhli — APPROVED** [review]

File: frontend/src/APIClients/AuthAPIClient.ts
Comment (sherryhli) [inline frontend/src/APIClients/AuthAPIClient.ts:46]: A small nitpick: I would leave the try-catch and the alert in the component since it's related to the UI. Don't need to address this right now though, I'll make a note to change it the next time we touch this section of the code 🙂

**Review by sherryhli — APPROVED** [review]
LGTM, 🚢 it!

**Pull request merged**
